hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableOutputFormat.java

```diff
@@ -460,27 +461,29 @@ public CarbonLoadModel getLoadModel() {
 
     private CarbonOutputIteratorWrapper[] iterators;
 
-    private int counter;
+    private AtomicLong counter;
 
     CarbonMultiRecordWriter(CarbonOutputIteratorWrapper[] iterators,
         DataLoadExecutor dataLoadExecutor, CarbonLoadModel loadModel, Future future,
         ExecutorService executorService) {
       super(null, dataLoadExecutor, loadModel, future, executorService);
       this.iterators = iterators;
+      counter = new AtomicLong(0);
     }
 
-    @Override public synchronized void write(NullWritable aVoid, ObjectArrayWritable objects)
+    @Override public void write(NullWritable aVoid, ObjectArrayWritable objects)
         throws InterruptedException {
-      iterators[counter].write(objects.get());
-      if (++counter == iterators.length) {
-        //round robin reset
-        counter = 0;
+      int hash = (int) (counter.incrementAndGet() % iterators.length);
```

Comment:
why not make counter AtomicInteger? Then no need to type cast to int

Reply:
If makes an integer and write called for more than INT_MAX records, it will give negative results,
So, keeping long is enough for very huge record. hence long.

But always long % int will be within int. so a safe type cast.

https://stackoverflow.com/questions/7262133/will-a-long-int-will-always-fit-into-an-int